core/domain/activity_validators.py

```diff
@@ -0,0 +1,90 @@
+# coding: utf-8
+#
+# Copyright 2020 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Validators for activity models."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+from constants import constants
+from core.domain import activity_domain
+from core.domain import base_model_validators
+from core.platform import models
+import feconf
+
+ERROR_CATEGORY_PROPERTY_FETCH_CHECK = 'fetch properties'
+
+(
+    collection_models, exp_models,) = (
+        models.Registry.import_models([
+            models.NAMES.collection, models.NAMES.exploration]))
+
+
+class ActivityReferencesModelValidator(
+        base_model_validators.BaseModelValidator):
+    """Class for validating ActivityReferencesModels."""
+
+    @classmethod
+    def _get_model_id_regex(cls, unused_item):
+        # Valid id: featured.
+        regex_string = '^(%s)$' % '|'.join(
+            feconf.ALL_ACTIVITY_REFERENCE_LIST_TYPES)
+        return regex_string
+
+    @classmethod
+    def _get_model_domain_object_instance(cls, item):
+        activity_references_list = []
+
+        try:
+            for reference in item.activity_references:
+                activity_references_list.append(
+                    activity_domain.ActivityReference(
+                        reference['type'], reference['id']))
+        except Exception as e:
+            cls._add_error(
+                ERROR_CATEGORY_PROPERTY_FETCH_CHECK,
+                'Entity id %s: Entity properties cannot be fetched completely '
+                'with the error %s' % (item.id, e))
+            return None
+
+        return activity_domain.ActivityReferences(activity_references_list)
+
+    @classmethod
+    def _get_external_id_relationships(cls, item):
+        exploration_ids = []
+        collection_ids = []
+
+        try:
+            for reference in item.activity_references:
+                if reference['type'] == constants.ACTIVITY_TYPE_EXPLORATION:
+                    exploration_ids.append(reference['id'])
+                elif reference['type'] == constants.ACTIVITY_TYPE_COLLECTION:
+                    collection_ids.append(reference['id'])
```

Comment:
should we add_error for `else` case?

Reply:
I don't think we should since there can be new activity types added later - and any issue with invalid type will already be caught while fetching properties in `_get_model_domain_object_instance` function.

Reply:
I think it would be a good idea to do it, otherwise when someone does add a new type, they'll not know that they have to add it here.

Reply:
Already verified via domain object validation, so test will fail either ways: https://github.com/oppia/oppia/pull/10416/files#diff-17cdad9697c37d27bb90ada4d971e5c9R133